Source/Core/DolphinQt/Debugger/CodeWidget.cpp

```diff
@@ -95,59 +96,72 @@ void CodeWidget::CreateWidgets()
   layout->setSpacing(0);
 
   m_search_address = new QLineEdit;
-  m_search_symbols = new QLineEdit;
   m_code_diff = new QPushButton(tr("Diff"));
   m_code_view = new CodeViewWidget;
 
   m_search_address->setPlaceholderText(tr("Search Address"));
-  m_search_symbols->setPlaceholderText(tr("Filter Symbols"));
 
   // Callstack
-  auto* callstack_box = new QGroupBox(tr("Callstack"));
-  auto* callstack_layout = new QVBoxLayout;
+  auto* callstack_widget = new QWidget;
+  auto* callstack_layout = new QGridLayout;
+  auto* callstack_label = new QLabel(tr("Callstack"));
+  m_search_callstack = new QLineEdit();
   m_callstack_list = new QListWidget;
 
-  callstack_box->setLayout(callstack_layout);
-  callstack_layout->addWidget(m_callstack_list);
+  callstack_widget->setLayout(callstack_layout);
+  callstack_layout->addWidget(callstack_label, 0, 0);
+  callstack_layout->addWidget(m_search_callstack, 0, 1);
+  callstack_layout->addWidget(m_callstack_list, 1, 0, -1, -1);
```

Comment:
Can't a helper function be used to deduplicate the code? Something like this might work:
```C++
m_box_splitter = new QSplitter(Qt::Vertical);

auto add_search_line_edit = [this](const QString& name, QListWidget* list_widget) {
  auto* widget = new QWidget;
  auto* layout = new QGridLayout;
  auto* label = new QLabel(name);
  auto* search_line_edit = new QLineEdit;

  widget->setLayout(layout);
  layout->addWidget(label, 0, 0);
  layout->addWidget(search_line_edit, 0, 1);
  layout->addWidget(list_widget, 1, 0, -1, -1);
  m_box_splitter->addWidget(widget);
  return search_line_edit;
}

// Callstack
m_callstack_list = new QListWidget;
m_search_callstack = add_search_line_edit(tr("Callstack"), m_callstack_list);

// Symbols
m_symbols_list = new QListWidget;
m_search_symbols = add_search_line_edit(tr("Symbols"), m_symbols_list);

// Function calls
m_function_calls_list = new QListWidget;
m_search_calls =  add_search_line_edit(tr("Calls"), m_function_calls_list);

// Function callers
m_function_callers_list = new QListWidget;
m_search_callers = add_search_line_edit(tr("Callers"), m_function_callers_list);
```

Reply:
Ooh that's some juicy code. I'm not good at writing stuff like this, but it works great.